pcdsdevices/state.py

```diff
@@ -69,15 +71,40 @@ def __init__(self, prefix, *, name, **kwargs):
             raise TypeError(('StatePositioner must be subclassed with at '
                              'least a state signal'))
         super().__init__(prefix, name=name, **kwargs)
-        self._valid_states = [state for state in self.states_list
-                              if state not in self._invalid_states
-                              and state is not None]
+        self._state_initialized = False
+        self._state_init_cbid = False
+        if self.states_list:
+            self._state_init()
+        else:
+            cbid = self.state.subscribe(self._late_state_init,
+                                        event_type=self._state_meta_sub,
+                                        run=False)
+            self._state_init_cbid = cbid
+
+    # @required_for_connection # This never made it into ophyd...
```

Comment:
It's there - just not yet in a release: https://github.com/bluesky/ophyd/blob/be880e1bcc87c05668bf300845e32416537e647b/ophyd/device.py#L1549 ☹️

Reply:
Yeah, that's what I meant. I wrote this thing out assuming things like `required_for_connection` and `SUB_META` were available and then I had to redo it.

That being said, this PR will be much easier to review after the other PR goes in because the other PR accidentally fixed a problem that made this one possible.

Reply:
Almost a year later, I have uncommented this as it is in the upcoming 1.4.0 and we're not waiting for it to come out of rcs